inloop/grading/models.py

```diff
@@ -1,14 +1,61 @@
+import os
+from shutil import make_archive
+
+from django.conf import settings
+from django.core.files.uploadedfile import SimpleUploadedFile
 from django.db import models
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+from django.utils import timezone
+from django.utils.timesince import timesince
 
 from inloop.solutions.models import Solution
 
 
+def get_upload_path(obj, filename):
+    """
+    Return an upload file path.
+
+    All Plagiarism Test zip files will be stored in a shared directory.
+    """
+    return "plagiarism_tests/{created_at}/{filename}".format_map({
+        "created_at": obj.created_at,
+        "filename": filename,
+    })
+
+
 class PlagiarismTest(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
     command = models.TextField(default="", help_text="Command that was used to perform the test")
+    zip_file = models.FileField(upload_to=get_upload_path, null=True)
```

Comment:
The migration is missing. Please also commit the migrations file that Django `makemigrations` generates, otherwise it will only work on your computer.

Reply:
Fixed.